pkg/kv/store_message.go

```diff
@@ -0,0 +1,60 @@
+package kv
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// StoreMessage protobuf generic implementation for kv.Store interface applicable for all data models
+type StoreMessage struct {
+	Store Store
+}
+
+func (s *StoreMessage) GetMsg(ctx context.Context, key []byte, msg protoreflect.ProtoMessage) error {
+	val, err := s.Store.Get(ctx, key)
+	if err != nil {
+		return err
+	}
+	return proto.Unmarshal(val, msg)
+}
+
+func (s *StoreMessage) SetMsg(ctx context.Context, key []byte, msg protoreflect.ProtoMessage) error {
+	val, err := proto.Marshal(msg)
+	if err != nil {
+		return err
+	}
```

Comment:
relevant for some places in this code - and probably in our storage too - the error should include more information like the key.